# HG changeset patch
# User Guillaume Flandin <guillaume.offline@gmail.com>
# Date 1547298880 18000
#      Sat Jan 12 08:14:40 2019 -0500
# Branch stable
# Node ID 75c7aafca6bb27ab564c8faf1e36b269eac6a21c
# Parent  0d3eca5955dcd4d1bcb0327d46ef558111bc7488
test for bug #54995

* test/bug-54995/module.mk, test/bug-54995/@testclass/subsref.m,
test/bug-54995/@testclass/testclass.m, test/bug-54995/bug-54995.tst:
New files.

* test/module.mk: Update.

diff --git a/test/bug-54995/@testclass/subsref.m b/test/bug-54995/@testclass/subsref.m
new file mode 100644
--- /dev/null
+++ b/test/bug-54995/@testclass/subsref.m
@@ -0,0 +1,3 @@
+function val = subsref (obj, S)
+  val = builtin ("subsref", obj, S);
+endfunction
diff --git a/test/bug-54995/@testclass/testclass.m b/test/bug-54995/@testclass/testclass.m
new file mode 100644
--- /dev/null
+++ b/test/bug-54995/@testclass/testclass.m
@@ -0,0 +1,4 @@
+function obj = testclass ()
+  obj = struct ("x", eye (4));
+  obj = class(obj, "testclass");
+endfunction
diff --git a/test/bug-54995/bug-54995.tst b/test/bug-54995/bug-54995.tst
new file mode 100644
--- /dev/null
+++ b/test/bug-54995/bug-54995.tst
@@ -0,0 +1,21 @@
+## Copyright (C) 2018 Guillaume Flandin
+##
+## This file is part of Octave.
+##
+## Octave is free software: you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+%!test <*54995>
+%! obj = testclass ();
+%! assert (obj.x(1), 1);
diff --git a/test/bug-54995/module.mk b/test/bug-54995/module.mk
new file mode 100644
--- /dev/null
+++ b/test/bug-54995/module.mk
@@ -0,0 +1,6 @@
+bug_54995_TEST_FILES = \
+  %reldir%/@testclass/testclass.m \
+  %reldir%/@testclass/subsref.m \
+  %reldir%/bug-54995.tst
+
+TEST_FILES += $(bug_54995_TEST_FILES)
diff --git a/test/module.mk b/test/module.mk
--- a/test/module.mk
+++ b/test/module.mk
@@ -70,16 +70,17 @@ include %reldir%/bug-50035/module.mk
 include %reldir%/bug-50716/module.mk
 include %reldir%/bug-51192/module.mk
 include %reldir%/bug-51532/module.mk
 include %reldir%/bug-51534/module.mk
 include %reldir%/bug-51599/module.mk
 include %reldir%/bug-52075/module.mk
 include %reldir%/bug-52722/module.mk
 include %reldir%/bug-53027/module.mk
+include %reldir%/bug-54995/module.mk
 include %reldir%/class-concat/module.mk
 include %reldir%/classdef/module.mk
 include %reldir%/classdef-multiple-inheritance/module.mk
 include %reldir%/classes/module.mk
 include %reldir%/ctor-vs-method/module.mk
 include %reldir%/fcn-handle-derived-resolution/module.mk
 include %reldir%/local-functions/module.mk
 include %reldir%/mex/module.mk
